autopilot/agent.go

```diff
@@ -448,20 +458,69 @@ func (a *Agent) controller(startingBalance btcutil.Amount) {
 					continue
 				}
 
-				nID := NewNodeID(chanCandidate.PeerKey)
-				pendingOpens[nID] = Channel{
-					Capacity: chanCandidate.ChanAmt,
-					Node:     nID,
-				}
-
 				go func(directive AttachmentDirective) {
-
+					// We'll start out by attempting to
+					// connect to the peer in order to begin
+					// the funding workflow.
 					pub := directive.PeerKey
-					err := a.cfg.ChanController.OpenChannel(
+					alreadyConnected, err := a.cfg.ConnectToPeer(
+						pub, directive.Addrs,
+					)
+					if err != nil {
+						log.Warnf("Unable to connect "+
+							"to %x: %v",
+							pub.SerializeCompressed(),
+							err)
+						return
```

Comment:
This should trigger a failed channel signal so we can recycle and attempt to connect to other peers.

Reply:
Should a new one be added or just reuse `OnChannelOpenFailure`?

Reply:
I think we can just re-use in this case.

Reply:
Fixed.

Reply:
As we can't connect to them we should also mark the node as failed in this instance. In the future however, we'll put them on a re-try queue.